src/rules/no_node_globals.rs

```diff
@@ -0,0 +1,308 @@
+use deno_ast::SourceRangedForSpanned;
+
+#[derive(Debug)]
+pub struct NoNodeGlobals;
+
+const CODE: &str = "no-node-globals";
+const MESSAGE: &str = "NodeJS globals are not available in Deno";
+const IMPORT_HINT: &str = "Import from the appropriate module instead";
+const IMPORT_FIX_DESC: &str = "Replace node global with module import";
+const REPLACE_FIX_DESC: &str = "Replace node global with corresponding value";
+const REPLACE_HINT: &str = "Use the corresponding value instead";
+
+static NODE_GLOBALS: phf::Map<&'static str, FixKind> = phf::phf_map! {
+  "process" => FixKind::Import { module: "node:process", import: "process" },
+  "Buffer" => FixKind::Import { module: "node:buffer", import: "{ Buffer }" },
+  "global" => FixKind::Replace("globalThis"),
+  "setImmediate" => FixKind::Import { module: "node:timers", import: "{ setImmediate }" },
+  "clearImmediate" => FixKind::Import { module: "node:timers", import: "{ clearImmediate }" },
+};
+
+impl LintRule for NoNodeGlobals {
+  fn lint_program_with_ast_view<'view>(
+    &self,
+    context: &mut Context<'view>,
+    program: Program<'view>,
+  ) {
+    NoNodeGlobalsHandler {
+      most_recent_import_range: None,
+    }
+    .traverse(program, context);
+  }
+
+  fn code(&self) -> &'static str {
+    CODE
+  }
+
+  fn tags(&self) -> &'static [&'static str] {
+    &["recommended"]
+  }
+
+  #[cfg(feature = "docs")]
+  fn docs(&self) -> &'static str {
+    include_str!("../../docs/rules/no_node_globals.md")
+  }
+}
+
+struct NoNodeGlobalsHandler {
+  most_recent_import_range: Option<SourceRange>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum FixKind {
+  Import {
+    module: &'static str,
+    import: &'static str,
+  },
+  Replace(&'static str),
+}
+
+#[derive(Default)]
+enum AddNewline {
+  Leading,
+  Trailing,
+  #[default]
+  None,
+}
+
+impl FixKind {
+  fn hint(&self) -> &'static str {
+    match self {
+      FixKind::Import { .. } => IMPORT_HINT,
+      FixKind::Replace(_) => REPLACE_HINT,
+    }
+  }
+  fn description(&self) -> &'static str {
+    match self {
+      FixKind::Import { .. } => IMPORT_FIX_DESC,
+      FixKind::Replace(_) => REPLACE_FIX_DESC,
+    }
+  }
+  fn to_text(self, newline: AddNewline) -> Cow<'static, str> {
+    match self {
+      FixKind::Import { module, import } => {
+        let (leading, trailing) = match newline {
+          AddNewline::Leading => ("\n", ""),
+          AddNewline::Trailing => ("", "\n"),
+          AddNewline::None => ("", ""),
+        };
+        format!("{leading}import {import} from \"{module}\";{trailing}").into()
+      }
+      FixKind::Replace(new_text) => new_text.into(),
+    }
+  }
+}
+
+fn program_code_start(program: Program) -> SourcePos {
+  match program_ref(program) {
+    ast_view::ProgramRef::Module(m) => {
+      m.body.first().map_or(program.start(), |node| node.start())
+    }
+    ast_view::ProgramRef::Script(s) => {
+      s.body.first().map_or(program.start(), |node| node.start())
```

Comment:
Can you switch these to `.map(...).unwrap_or(...)`? https://github.com/denoland/deno/pull/18212